### PR TITLE
feat: [IDP-135]: Feature build pipeline changes

### PR DESCRIPTION
**Pull request by devesh-harness**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/43344)

### CODE REVIEW

**Comment by devesh-harness** [comment]
trigger messagecheck

**Review by vikyathharekal — APPROVED** [review]
> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43344)* status: 2 file(s) to review. 1 of 2 approvals obtained.

File: idp-service/container/scripts/start_process.sh [hunk 1]
```diff
@@ -19,22 +20,12 @@ fi
...
 fi
 
-if [[ "${ENABLE_SERIALGC}" == "true" ]]; then
-    export GC_PARAMS=" -XX:+UseSerialGC -Dfile.encoding=UTF-8"
-else
-    export GC_PARAMS=" -XX:+UseG1GC -XX:InitiatingHeapOccupancyPercent=40 -XX:MaxGCPauseMillis=1000 -Dfile.encoding=UTF-8"
-fi
-
-export JAVA_OPTS="-Xmx${MEMORY} -XX:+HeapDumpOnOutOfMemoryError -Xloggc:mygclogfilename.gc $GC_PARAMS $JAVA_ADVANCED_FLAGS"
+export GC_PARAMS=" -XX:+UseG1GC -XX:InitiatingHeapOccupancyPercent=40 -XX:MaxGCPauseMillis=1000 -Dfile.encoding=UTF-8"
```
Comment (sandeepa-kv-jois) [inline idp-service/container/scripts/start_process.sh:26]: Is these the expected GC_PARAMS values?

**Review by sandeepa-kv-jois — APPROVED** [review]

File: idp-service/container/scripts/start_process.sh (hunk 1 above)
Comment (devesh-harness) [reply idp-service/container/scripts/start_process.sh:26]: yes checked in other services they are using same value.

**Review by deepharness — APPROVED** [review]

**Review by deepharness — APPROVED** [review]
> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43344)* status: 1. unresolved discussions.

**Review by vikyathharekal — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43344)* status: 1. unresolved discussions. 

___
*[`idp-service/container/scripts/start_process.sh` line 26 at r1](https://reviewable.io/reviews/harness/harness-core/43344#-NNBL5rnrvMPpR2OVWn8-r1-26:-NNHVTkHAhu0nZR8Xs4B:bg7igkv) ([raw file](https://github.com/harness/harness-core/blob/f6d3cacee40e2cd13900d92a1359fecc70c54ee2/idp-service/container/scripts/start_process.sh#L26)):*
<details><summary><i>Previously, devesh-harness wrote…</i></summary><blockquote>

yes checked in other services they are using same value.
</blockquote></details>

:lgtm:

**Review by sandeepa-kv-jois — APPROVED** [review]

**Review by sandeepa-kv-jois — COMMENTED** [review]
> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43344)* status: 1. unresolved discussions.

**Review by sandeepa-kv-jois — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43344)* status: :shipit: complete! ✓

**Pull request merged**